agent/taskresource/volume/dockervolume.go

```diff
@@ -56,6 +55,12 @@ type VolumeResource struct {
 	statusToTransitions map[resourcestatus.ResourceStatus]func() error
 	client              dockerapi.DockerClient
 	ctx                 context.Context
+
+	// terminalReason should be set for resource creation failures. This ensures
+	// the resource object carries some context for why provisoning failed.
+	terminalReason     string
+	terminalReasonOnce sync.Once
```

Comment:
Whats the rationale for including the sync.Once?

Reply:
hm it was in the existing codepaths that populated the terminalReason string. need to dig into the old ones to figure out why we added that in the first place 🤔

Reply:
sigh, noting outside of `This field does not accept updates.` [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task.go#L1822)

im thinking since its a public setter for terminal reason, we didn't want the field clobbered once it's been set